File: src/main/java/io/vproxy/vfx/ui/stage/VStageInitParams.java
package io.vproxy.vfx.ui.stage;

import io.vproxy.vfx.ui.scene.VScene;

public class VStageInitParams {
    public boolean closeButton = true;
    public boolean maximizeAndResetButton = true;
    public boolean iconifyButton = true;
    public boolean resizable = true;
    public VScene initialScene = null;

    public VStageInitParams() {
    }

    public VStageInitParams setCloseButton(boolean closeButton) {
        this.closeButton = closeButton;
        return this;
    }

    public VStageInitParams setMaximizeAndResetButton(boolean maximizeAndResetButton) {
        this.maximizeAndResetButton = maximizeAndResetButton;
        return this;
    }

    public VStageInitParams setIconifyButton(boolean iconifyButton) {
        this.iconifyButton = iconifyButton;
        return this;
    }

    public VStageInitParams setResizable(boolean resizable) {
        this.resizable = resizable;
        return this;
    }

    public VStageInitParams setInitialScene(VScene initialScene) {
        this.initialScene = initialScene;
        return this;
    }
}

File: src-ts/src/ui/stage/VStageInitParams.ts
import type { VScene } from '../scene/VScene.js';

export class VStageInitParams {
  closeButton = true;
  maximizeAndResetButton = true;
  iconifyButton = true;
  resizable = true;
  initialScene: VScene | null = null;

  setCloseButton(v: boolean): this {
    this.closeButton = v;
    return this;
  }
  setMaximizeAndResetButton(v: boolean): this {
    this.maximizeAndResetButton = v;
    return this;
  }
  setIconifyButton(v: boolean): this {
    this.iconifyButton = v;
    return this;
  }
  setResizable(v: boolean): this {
    this.resizable = v;
    return this;
  }
  setInitialScene(s: VScene): this {
    this.initialScene = s;
    return this;
  }
}
